src/scripts/main.js

```diff
@@ -1,3 +1,449 @@
+
+  scoreValue = 0;
+  updateScore();
+  messageStart();
+}
+
+button.addEventListener('click', () => {
+  if (button.classList.contains('start')) {
+    gameStart();
+  } else {
+    // button.classList.remove('restart');
+    // button.classList.add('start');
+    // button.textContent = 'Start';
+
+    restart();
+    updateGameFields();
+    appearingOneRandomCell();
+    updateGameFields();
+    appearingOneRandomCell();
+    updateGameFields();
+  }
+});
+
+function getRandonCell() {
+  const emptyCells = [];
+
+  table.forEach((row, rowIndex) => {
+    row.forEach((cell, columnIndex) => {
+      if (cell === 0) {
+        emptyCells.push([rowIndex, columnIndex]);
+      }
+    });
+  });
+
+  return emptyCells[Math.floor(Math.random() * (emptyCells.length - 1))];
+}
+
+function left(paramArr) {
+  const arr = paramArr.filter(item => item !== 0);
+
+  for (let i = 0; i < arr.length - 1; i++) {
+    const current = arr[i];
+    const next = arr[i + 1];
+
+    if (current === next) {
+      arr[i] = current * 2;
+      scoreValue += arr[i];
+      updateScore();
+      arr[i + 1] = 0;
+    }
+  }
+
+  const mergedArr = arr.filter(item => item !== 0);
+
+  paramArr[0] = mergedArr[0] || 0;
+  paramArr[1] = mergedArr[1] || 0;
+  paramArr[2] = mergedArr[2] || 0;
+  paramArr[3] = mergedArr[3] || 0;
+}
+
+function right(paramArr) {
+  const arr = paramArr.filter(item => item !== 0);
+
+  for (let i = arr.length - 1; i >= 0; i--) {
+    const current = arr[i];
+    const prev = arr[i - 1];
+
+    if (current === prev) {
+      arr[i] = current * 2;
+      scoreValue += arr[i];
+      updateScore();
+      arr[i - 1] = 0;
+    }
+  }
+
+  const mergedArr = arr.filter(item => item !== 0);
+
+  while (mergedArr.length < 4) {
+    mergedArr.unshift(0);
+  }
+
+  paramArr[0] = mergedArr[0] || 0;
+  paramArr[1] = mergedArr[1] || 0;
+  paramArr[2] = mergedArr[2] || 0;
+  paramArr[3] = mergedArr[3] || 0;
+}
+
+function moveLeft() {
+  const json = JSON.stringify(table);
+  const copyTable = JSON.parse(json);
+
+  for (const row of table) {
+    left(row);
+  }
+
+  for (let i = 0; i < 16; i++) {
+    const rowIndex = Math.floor(i / 4);
+    const colIndex = i - (rowIndex * 4);
+
+    const tableValue = table[rowIndex][colIndex];
+    const copyTableValue = copyTable[rowIndex][colIndex];
+
+    if (tableValue !== copyTableValue) {
+      appearingOneRandomCell();
+      updateGameFields();
+
+      break;
+    }
+  }
+}
+
+function moveRight() {
+  const json = JSON.stringify(table);
+  const copyTable = JSON.parse(json);
+
+  for (const row of table) {
+    right(row);
+  }
+
+  for (let i = 0; i < 16; i++) {
+    const rowIndex = Math.floor(i / 4);
+    const colIndex = i - (rowIndex * 4);
+
+    const tableValue = table[rowIndex][colIndex];
+    const copyTableValue = copyTable[rowIndex][colIndex];
+
+    if (tableValue !== copyTableValue) {
+      appearingOneRandomCell();
+      updateGameFields();
+
+      break;
+    }
+  }
+}
```

Comment:
for moveLeft and moveRight also almost same code and can be combined in one function.
or at least loop same for four move functions and can be writed in separate function